code/lib/core-server/src/utils/StoryIndexGenerator.ts

```diff
@@ -346,8 +346,12 @@ export class StoryIndexGenerator {
       });
     }
 
+    const entriesWithoutDocsOnlyStories = entries.filter(
+      (entry) => !(entry.type === 'story' && entry.tags.includes('docsOnly'))
```

Comment:
Maybe this should be something more obscure like `stories-mdx-docsOnly`? Seems unfortunately to reserve a tags meaning for this deprecated reason.